src/components/ChartSetting.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import Menu from './Menu.jsx';
+import { connect } from '../store/Connect';
+import { 
+    EditIcon,
+    PositionLeftIcon,
+    PositionBottomIcon,
+    ThemeDarkIcon,
+    ThemeLightIcon,
+    ChevronRightIcon,
+    BackIcon,
+    FlagIcons
+} from './Icons.jsx';
+import '../../sass/_ciq-chart-setting.scss';
+
+
+const ChartSetting = ({
+    Menu,
+    menuOpen,
+    languages,
+    setView,
+    view,
+    setLanguage
+}) => {
+    const renderMain = () => {
+        return <div>
+            <div className='title'> {t.translate('Settings')} </div>
+            <div className='body'>
+                <div className="ciq-list-setting">
+                    {/*
+                        other chart setting option which not use for now
+                    <div className="ciq-item">
+                        <span className="ciq-icon-text">{t.translate('Position')}</span>
+                        <div className="ciq-action">
+                            <PositionLeftIcon 
+                            />
+                            <PositionBottomIcon 
+                            />
+                        </div>
+                    </div>
+                    <div className="ciq-item">
+                        <span className="ciq-icon-text">{t.translate('Theme')}</span>
+                        <div className="ciq-action">
+                            <ThemeDarkIcon 
+                            />
+                            <ThemeLightIcon 
+                            />
+                        </div>
+                    </div>*/}
+                    <div className="ciq-item ciq-list-item-lng">
+                        <span className="ciq-icon-text">{t.translate('Language')}</span>
+                        <div className="ciq-action">
+                            <span></span>
+                            <ChevronRightIcon 
+                                onClick={ () => setView('language') }
+                                />
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    }
+    const renderLanguage = () =>{
+        return <div>
+            <div className='title'>
+                <BackIcon
+                    onClick={() => setView() }
+                    />
+                {t.translate('Language')}
+            </div>
+            <div className='body'>
+                <div className="ciq-list ciq-list-language">
+                    {languages.map( (language,index) => {
+                        return <div className="ciq-list-item" key={index} >
+                            <span onClick={()=> setLanguage(language.key) }>
+                                {language.icon}
+                            </span>
+                            <span className="ciq-icon-text">{t.translate(language.name)}</span>
```

Comment:
I think It need, because For example, we write `Malaysia` in English but its `مالزی` in Farsi.
I can remove it, if you see it's not necessary. I just try to analyze project as I can.